src/cc.h

```diff
@@ -1,14 +1,12 @@
-#pragma once
-
 #ifndef CC_H_
 #define CC_H_
 
 #include <string>
 #include <vector>
 
 namespace cc {
-    std::vector<std::string> chrToVec(char const *chrArr[], int argc);
-    std::string collectVecInStr(std::vector<std::string> vec, size_t startIdx);
+    std::vector<std::string> chrToVec(char const *chrArr[], const int& argc);
```

Comment:
for simple types, you can still pass them as value, it won't be slower.

If you want to have some static analysis, I would recommand you to use [clang-tidy](https://clang.llvm.org/extra/clang-tidy/) which has a bunch of rules to help improving the code. You should remove the ones you don't agree with (for example, coding style).

An example on how to set it up in github: https://github.com/petitlapin/Li-Ri/pull/19/files. The .clang-tidy defines the rules to use (* for all, "minus" the rules you don't want) and the workflow just below allows github to run it on each pull request. The other files in the PR are changes I've applied with clang-tidy on my own code.

Another tool, maybe easier to use is cppcheck.

Note that you can run them locally, no need to do it on github if you don't want.